### PR TITLE
mds: remove the duplicated or incorrect respond

### PR DESCRIPTION
**Pull request by lxbsz**
The try_get_auth_inode() has already responded to the client or
forwarded to auth MDS if returning NULL.

Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mds/Server.cc [hunk 1]
```diff
@@ -2529,7 +2529,6 @@ void Server::dispatch_client_request(MDRequestRef& mdr)
   if (is_full) {
     CInode *cur = try_get_auth_inode(mdr, req->get_filepath().get_ino());
     if (!cur) {
-      respond_to_request(mdr, -EINVAL);
```
Comment (batrick) [inline src/mds/Server.cc:2532]: Let's add a comment that the request is already responded to!

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/mds/Server.cc (hunk 1 above)
Comment (batrick) [reply src/mds/Server.cc:2532]: (great catch @lxbsz !)

Comment (lxbsz) [reply src/mds/Server.cc:2532]: Sorry, I missed this comment, I will update it. Thanks @batrick

**Comment by lxbsz** [comment]
@batrick Done. Thanks.

**Review by batrick — APPROVED** [review]

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by vshankar** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2022-Jan-13

**Pull request merged**